nms/app/packages/magmalte/app/state/feg/SubscriberState.js

```diff
@@ -0,0 +1,94 @@
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import type {EnqueueSnackbarOptions} from 'notistack';
+import type {network_id, subscriber_state} from '@fbcnms/magma-api';
+
+import {FetchSubscriberState} from '../lte/SubscriberState';
+import {getServicedAccessNetworks} from '../../components/FEGServicingAccessGatewayKPIs';
+/**
+ * Props passed when initializing Subscriber State.
+ *
+ * @property {network_id} networkId Id of the federation network.
+ * @property {({[string]: federation_gateway}) => void} setSessionState Sets the subscriber session state.
+ * @property {(msg, cfg,) => ?(string | number),} enqueueSnackbar Snackbar to display error.
+ */
+type InitSubscriberStateProps = {
+  networkId: network_id,
+  setSessionState: ({[network_id]: {[string]: subscriber_state}}) => void,
+  enqueueSnackbar?: (
+    msg: string,
+    cfg: EnqueueSnackbarOptions,
+  ) => ?(string | number),
+};
+
+/**
+ * Initializes the Subscriber State with all the sessions that
+ * this federation network services.
+ *
+ * @param {InitSubscriberStateProps} props an object containing the neccessary values to initalize the subscriber state
+ */
+export async function InitSubscriberState(props: InitSubscriberStateProps) {
+  const {networkId, setSessionState, enqueueSnackbar} = props;
+  const sessionState = await FetchFegSubscriberState({
+    networkId,
+    enqueueSnackbar,
+  });
+  setSessionState(sessionState);
+}
+
+/**
+ * Props passed when fetching subscriber state.
+ *
+ * @param {network_id} networkId Id of the federation network.
+ * @param {(msg, cfg,) => ?(string | number),} enqueueSnackbar Snackbar to display error.
+ */
+type FetchProps = {
+  networkId: string,
+  enqueueSnackbar?: (
+    msg: string,
+    cfg: EnqueueSnackbarOptions,
+  ) => ?(string | number),
+};
+
+/**
+ * Fetches and returns the subscriber session state of all the serviced
+ * federated lte networks under by this federation network and whose
+ * subscriber session is not managed by the HSS.
+ *
+ * @param {FetchProps} props an object containing the network id and snackbar to display error.
+ */
+export async function FetchFegSubscriberState(props: FetchProps) {
+  const {networkId, enqueueSnackbar} = props;
+  const servicedAccessNetworks = await getServicedAccessNetworks(
+    networkId,
+    enqueueSnackbar,
+  );
+  const sessionState = {};
+  for (const servicedAccessNetwork of servicedAccessNetworks) {
+    // only save subscriber state of networks with hss disabled
+    if (!servicedAccessNetwork?.cellular?.epc?.hss_relay_enabled) {
```

Comment:
Yes  that is intended. But we also can guarantee that all of those 3 property won't be missing because they are required properties. So, I have removed the question marks in order to remove confusions.

Reply:
Nvm, I have removed that line because we have to display when the network is using HSS too.